### ISSUE TITLE
URL in config files are not consistent with URL in Model Zoo

### ISSUE THREAD

**Issue by gadcam** [facebookresearch/Detectron#326]
### Expected results
Use the model <code>X-152-32x8d-FPN-IN5k</code> to do inference.
I can do successfuly inference on <code>e2e_keypoint_rcnn_R-50-FPN_s1x</code> for example.

### Actual results

It seems there is a problem when building the network in the function <code>add_topdown_lateral_module</code> of <code>/lib/modeling/FPN.py</code>
So this piece of code
```python
    lat = model.Conv(
        fpn_lateral,
        fpn_bottom + '_lateral',
        dim_in=dim_lateral,
        dim_out=dim_top,
        kernel=1,
        pad=0,
        stride=1,
        weight_init=(
            const_fill(0.0) if cfg.FPN.ZERO_INIT_LATERAL
            else ('XavierFill', {})
        ),
        bias_init=const_fill(0.0)
    )
```
The output of the command is the following:

```shell
terminate called after throwing an instance of 'caffe2::EnforceNotMet'
  what():  [enforce fail at blob.h:97] IsType<T>(). wrong type for the Blob instance.
  Blob contains nullptr (uninitialized) while caller expects caffe2::Tensor<caffe2::CUDAContext> .
Offending Blob name: gpu_0/fpn_inner_res2_2_sum_lateral_w.
Error from operator:
	input: "gpu_0/res2_2_sum"
	input: "gpu_0/fpn_inner_res2_2_sum_lateral_w"
	input: "gpu_0/fpn_inner_res2_2_sum_lateral_b"
	output: "gpu_0/fpn_inner_res2_2_sum_lateral"
	name: ""
	type: "Conv"
      arg { name: "kernel" i: 1 }
      arg { name: "order" s: "NCHW" }
      arg { name: "pad" i: 0 }
      arg { name: "stride" i: 1 }
      arg { name: "exhaustive_search" i: 0 }
	device_option { device_type: 1 cuda_gpu_id: 0 }
	engine: "CUDNN"
	debug_info:
      [...]      
      File \"%DETECTRON_DIR%/lib/core/test_engine.py\", line 314, in initialize_model_from_cfg
      	model = model_builder.create(cfg.MODEL.TYPE, train=False, gpu_id=gpu_id)
      File \"%DETECTRON_DIR%/lib/modeling/model_builder.py\", line 125, in create
      	return get_func(model_type_func)(model)
      File \"%DETECTRON_DIR%/lib/modeling/model_builder.py\", line 90, in generalized_rcnn
      	freeze_conv_body=cfg.TRAIN.FREEZE_CONV_BODY
      File \"%DETECTRON_DIR%/lib/modeling/model_builder.py\", line 231, in build_generic_detection_model
      	optim.build_data_parallel_model(model, _single_gpu_build_func)
      File \"%DETECTRON_DIR%/lib/modeling/optimizer.py\", line 54, in build_data_parallel_model
      	single_gpu_build_func(model)
      File \"%DETECTRON_DIR%/lib/modeling/model_builder.py\", line 171, in _single_gpu_build_func
      	blob_conv, dim_conv, spatial_scale_conv = add_conv_body_func(model)
      File \"%DETECTRON_DIR%/lib/modeling/FPN.py\", line 77, in add_fpn_ResNet152_conv5_body
      	model, ResNet.add_ResNet152_conv5_body, fpn_level_info_ResNet152_conv5
      File \"%DETECTRON_DIR%/lib/modeling/FPN.py\", line 105, in add_fpn_onto_conv_body
      	model, fpn_level_info_func()
      File \"%DETECTRON_DIR%/lib/modeling/FPN.py\", line 166, in add_fpn
      	fpn_dim_lateral[i + 1]       # lateral input dimension
      File \"%DETECTRON_DIR%/lib/modeling/FPN.py\", line 244, in add_topdown_lateral_module
      	bias_init=const_fill(0.0)
      File \"/opt/caffe2/lib/python3.6/site-packages/caffe2/python/cnn.py\", line 112, in Conv
      	**kwargs
      File \"/opt/caffe2/lib/python3.6/site-packages/caffe2/python/brew.py\", line 121, in scope_wrapper
      	return func(*args, **new_kwargs)
      File \"/opt/caffe2/lib/python3.6/site-packages/caffe2/python/helpers/conv.py\", line 196, in conv
      	group, transform_inputs, **kwargs)
      File \"/opt/caffe2/lib/python3.6/site-packages/caffe2/python/helpers/conv.py\", line 149, in _ConvBase
      	**kwargs)
      File \"/opt/caffe2/lib/python3.6/site-packages/caffe2/python/core.py\", line 2047, in <lambda>
      	op_type, *args, **kwargs)
      File \"/opt/caffe2/lib/python3.6/site-packages/caffe2/python/core.py\", line 2024, in _CreateAndAddToSelf
      	op = CreateOperator(op_type, inputs, outputs, **kwargs)
*** Aborted at 1522258832 (unix time) try "date -d @1522258832" if you are using GNU date ***
PC: @     0x7f79d9908660 __GI_raise
*** SIGABRT (@0x93350000464a) received by PID 17994 (TID 0x7f78eb1fd700) from PID 17994; stack trace: ***
    @     0x7f79da3ee1b0 (unknown)
    @     0x7f79d9908660 __GI_raise
    @     0x7f79d9909c41 __GI_abort
    @     0x7f79cd47d025 __gnu_cxx::__verbose_terminate_handler()
    @     0x7f79cd47ac16 (unknown)
    @     0x7f79cd47ac61 std::terminate()
    @     0x7f79cd4a717e (unknown)
    @     0x7f79da3e350b start_thread
    @     0x7f79d99c916f __GI___clone
Aborted (core dumped)
```
### Detailed steps to reproduce
Run this command
```shell
python3 tools/infer_simple.py \
--cfg configs/12_2017_baselines/e2e_mask_rcnn_X-152-32x8d-FPN-IN5k_1.44x.yaml \
--wts  https://s3-us-west-2.amazonaws.com/detectron/ImageNetPretrained/25093814/X-152-32x8d-IN5k.pkl \
--output-dir $OUTPUT_DIR \
--image-ext png \
$INPUTDIR
```
### System information

* Operating system: Fedora 27
* Compiler version: ?
* CUDA version:  9.0.176
* cuDNN version: 7.0.5
* NVIDIA driver version: 390.25
* GPU models: Tesla K80
* `PYTHONPATH` environment variable: <code>/opt/caffe2/lib/python3.6/site-packages:$DETECTRONDIR</code>
* `python --version` output: Python 3.6.4

**Comment by gadcam** [comment]
I managed to run this model using the URL in the [Model Zoo](https://github.com/facebookresearch/Detectron/blob/master/MODEL_ZOO.md)
```
https://s3-us-west-2.amazonaws.com/detectron/37129812/12_2017_baselines/e2e_mask_rcnn_X-152-32x8d-FPN-IN5k_1.44x.yaml.09_35_36.8pzTQKYK/output/train/coco_2014_train%3Acoco_2014_valminusminival/generalized_rcnn/model_final.pkl
```


When the URL in the cfg file is
```
https://s3-us-west-2.amazonaws.com/detectron/ImageNetPretrained/25093814/X-152-32x8d-IN5k.pkl
```
This behavior does not seem to be specific to this model.

What is the difference between these two models ? The second URL should work right ?

**Comment by rbgirshick** [comment]
The first one (`model_final.pkl`) is the trained detection model, which should be used as the value for `TEST.WEIGHTS` or the argument to `--wts` in `tools/infer_simple.py`. The second one (`X-152-32x8d-IN5k.pkl`) is simply the ImageNet pretrained model that has not been finetuned for detection or had its structure augmented for detection (it's just a classification model). It's used as the value of `TRAIN.WEIGHTS`, since it specifies the initial weights used when training.

Note that we do not officially support `python3` yet, so there may be other issues related to that.

**Comment by gadcam** [comment]
Thank you for your explanation.
So far using Python 3 did not seem to cause any issue.
What do think about adding a message to tell the user the network structure is not the one described in the CFG to explain this ?
```
  what():  [enforce fail at blob.h:97] IsType<T>(). wrong type for the Blob instance.
  Blob contains nullptr (uninitialized) while caller expects caffe2::Tensor<caffe2::CUDAContext> .
Offending Blob name: gpu_0/fpn_inner_res2_2_sum_lateral_w.
```
If you give me some indication on how to do it, I can write the code and do a pull request.

**Comment by gadcam** [comment]
For future reference:

My suggestion is irrelevant as the error is occurring in Caffe2.

I had an info message which is showing something is wrong when <code>infer_engine.initialize_model_from_cfg</code> is used, which should be enough.

```log
INFO net.py:  57: Loading weights from: /tmp/detectron-download-cache/ImageNetPretrained/25093814/X-152-32x8d-IN5k.pkl
INFO net.py:  89: fpn_inner_res5_2_sum_w not found
INFO net.py:  89: fpn_inner_res5_2_sum_b not found
INFO net.py:  89: fpn_inner_res4_35_sum_lateral_w not found
INFO net.py:  89: fpn_inner_res4_35_sum_lateral_b not found
...
```
However, I can not figure why but the log is misleading at it shows an error in <code>initialize_model_from_cfg</code> when the error is happening with <code> im_detect_all</code>.

**Comment by smoosbau** [comment]
Hello together,

I'm trying to finetune the mask R-CNN at the moment. Training is running and everything is fine but after reading through this issue I'm asking myself if it's better to use the "standard" TRAIN.WEIGHTS for finetuning or to use weights already tuned for detection.
While training I'm freezing the complete backbone and retrain only the bbox_pred and cls_score layers. So do you have any thoughts which weights might be better to use?

Thanks in advance!
